lib/uplink/uplink.go

```diff
@@ -62,6 +63,14 @@ type Config struct {
 		// PartnerID is the identity given to the partner for value
 		// attribution
 		PartnerID string
+
+		// DialTimeout is the maximum time to wait connecting to another node.
+		// If not set, the library default (20 seconds) will be used.
+		DialTimeout time.Duration
+
+		// RequestTimeout is the maximum time to wait for a request response from another node.
+		// If not set, the library default (20 seconds) will be used.
+		RequestTimeout time.Duration
```

Comment:
I'm not sure what the expected behavior is but I just want to make sure this is what should be happening. Unless we add these fields to the `UplinkConfig` C struct, a C api user won't have a way to change them. This is fine so long as the default is correct; which, in this case, would be 20 seconds [(`transport.Default...`)](https://github.com/storj/storj/blob/8836fd6e469631e176cef5d05254cd253a9ddb87/pkg/transport/common.go) as per [`Config#setDefaults`](https://github.com/storj/storj/blob/8836fd6e469631e176cef5d05254cd253a9ddb87/lib/uplink/uplink.go#L94).

Reply:
We discussed and decided to put the C changes in another PR.